src/core_modules/capture-core/components/Pages/Search/SearchPage.types.js

```diff
@@ -11,21 +11,20 @@ export type SearchGroups = Array<{|
 
 export type SelectedSearchScopeId = ?string
 
-export type AvailableSearchOptions = $ReadOnly<{
-    [elementId: string]: {|
+export type AvailableSearchOption = {|
       +searchOptionId: string,
       +searchOptionName: string,
       +TETypeName: ?string,
-      +searchGroups: SearchGroups |}
-  }>
+      +searchGroups: SearchGroups
+|}
 
 export type ContainerProps = $ReadOnly<{|
   cleanSearchRelatedInfo: ()=>void,
   navigateToMainPage: ()=>void,
   showInitialSearchPage: ()=>void,
   navigateToRegisterUser: ()=>void,
   trackedEntityTypeId: string,
-  availableSearchOptions: AvailableSearchOptions,
+  availableSearchOption: ?AvailableSearchOption,
```

Comment:
Hey,
The user is currently able to search from the main page for tracker programs with 'Display front page list' set to false. Meaning that the `availableSearchOption` variable shouldn't be optional and that the same changes you did in `SearchPage.container` should be applied to the [MainPage.container](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/Pages/MainPage/MainPage.container.js#L131). 

Thanks!

Reply:
Oh, that's a good catch! Changed so we also use the hook in the mainpage. I also agree that `availableSearchOption` should be required, but how do we feel like we should handle the async-nature of this hook? I'm guessing we're only getting more async hooks if we are taking data directly from IndexedDB. Should we try to incorporate all loading states with the `ready`-prop? If so, Flow might still argue for it being undefined. Another way would be to allow undefined values, but return something appropriate in the component. WDYT?

Reply:
I'm more in favor of allowing undefined values and letting the component handle them instead of using `withLoadingIndicator` HOC and `ready` prop. Depending on the scenarios, the `Example.component.js` can make use of the different helpers from [LoadingMask](https://github.com/dhis2/capture-app/tree/master/src/core_modules/capture-core/components/LoadingMasks) and decide what is shown to the user.

Reply:
> Should we try to incorporate all loading states with the ready-prop?
>> I'm more in favor of allowing undefined values and letting the component handle them instead of using withLoadingIndicator HOC and ready prop. Depending on the scenarios, the Example.component.js can make use of the different helpers from LoadingMask and decide what is shown to the user.

In general, I think it depends (and there is no one size fits all solution). If we're only waiting for something from IndexedDB it probably makes sense to not show a loading spinner at all (and this is maybe best dealt with in the component). Other times, it might make sense to incorporate it into to the `ready` state (simplifying the component).

Reply:
Hey again,
After a discussion with Joakim we've ending up commenting out the flow issue here. We'll see if we can look into the HOC only returning the correct props using generics in the future, but not critical as of now.